### ISSUE TITLE
@babel/preset-typescript and generic arrow function (<T extends {}>)

### ISSUE THREAD

**Issue by Tsury** [babel/babel#9086]
## Bug Report

**Current Behavior**
I have an anonymous generic arrow function which looks like this:
`const myFunc = <T extends {}>(params: any): any => { }`

I'm using `vscode` and everything is fine, but when I try to build, I get this error from `babel-loader`:
`Unexpected token, expected ";"`

**Input Code**
- REPL or Repo link if applicable:

```js
`const myFunc = <T extends {}>(params: any): any => { }`
```

**Expected behavior/code**
Should just build my code

**Babel Configuration (.babelrc, package.json, cli command)**

My `.babelrc` has this global preset:
```
  "presets": [
    "@babel/preset-typescript"
  ],
```

and this env preset:
`["@babel/preset-react", { "development": true }]`

These global plugins:
```
    ["@babel/plugin-proposal-decorators", { "legacy" : true }],
    ["@babel/plugin-proposal-class-properties", { "loose" : true }],
    "@babel/plugin-syntax-dynamic-import",
    ["import", { "libraryName": "antd", "style": true }]
``` 

And these env plugins:
```
        ["@babel/plugin-transform-react-display-name"],
        ["emotion", { "sourceMap": true, "autoLabel": true }]
```
`babel-loader` in webpack has no additional settings to the `.babelrc` file.

**Environment**
- Babel version(s): 7.1.6
- Node/npm version: Node 11.0.0/npm 6.2.0
- OS: Windows 10
- How you are using Babel: loader

**Possible Solution**

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by nicolo-ribaudo** [comment]
This seems to work on master

**Comment by lapto** [comment]
Have the same issue, we had to return to react-native-typescript-transformer.

#7504 is probably related.

**Comment by danez** [comment]
I'm not able to reproduce this with 7.1.6 in the repl.

**Comment by lapto** [comment]
@danez Im running it using react-native. Seems like metro is stingy with its stacktrace but here it is:

`export const test = async <P extends IP>(body: IE<P>): Promise<void> => {`

`  
    at _class.raise (*/node_modules/@babel/parser/lib/index.js:4028:15)
    at _class.unexpected (*/node_modules/@babel/parser/lib/index.js:5343:16)
    at _class.semicolon (*/node_modules/@babel/parser/lib/index.js:5327:40)
    at _class.parseVarStatement (*/node_modules/@babel/parser/lib/index.js:7729:10)
    at _class.parseStatementContent (*/node_modules/@babel/parser/lib/index.js:7327:21)
    at _class.parseStatementContent (*/node_modules/@babel/parser/lib/index.js:10082:58)
    at _class.parseStatement (*/node_modules/@babel/parser/lib/index.js:7277:17)
    at _class.parseExportDeclaration (*/node_modules/@babel/parser/lib/index.js:8365:17)
    at _class.parseExportDeclaration (*/node_modules/@babel/parser/lib/index.js:10200:68)
    at _class.parseExport (*/node_modules/@babel/parser/lib/index.js:8312:31)
`

 I checked the version written in package.json of node_mules/@babel and its "version": "7.1.6",

**Comment by e-r-w** [comment]
If it helps, I've come across this issue too:

```ts
const someFunc = () => Promise.resolve(1);

export const worksFine = async <T>(thing: T): Promise<T> => {
  return someFunc().then(() => thing);
};

export const itsBroken = async <T>(thing: T): Promise<T> => {
  return await someFunc().then(() => thing);
};
```

Seems to break when you have the `await` keyword mixed in with generic arrow functions. @bable/core 7.1.6.

**Comment by danez** [comment]
The problem with typescript and async/await has been fixed in #9055 and will be released with 7.2.0.

**Comment by danez** [comment]
Please try again. It should work now with 7.2.0, as we fixed several issues mentioned in this issue.

If there are still problems please open new issues for them. Thank you.
